apps/src/applab/AppLabTooltipOverlay.jsx

```diff
@@ -0,0 +1,148 @@
+/** @file App Lab-specific Tooltip Overlay */
+
+import TooltipOverlay, {coordinatesProvider} from '../templates/TooltipOverlay';
+import elementUtils from './designElements/elementUtils';
+import { scaledDropPoint } from './gridUtils';
+import { connect } from 'react-redux';
+import { ApplabInterfaceMode } from './constants';
+import { ellipsify } from '../utils';
+
+var ELEMENT_ID_TEXT_MAX_CHAR = 12;
+
+const AppLabTooltipOverlay = React.createClass({
+  propTypes: {
+    // width, height, mouseX and mouseY are given in app-space, not screen-space
+    width: React.PropTypes.number,
+    height: React.PropTypes.number,
+    mouseX: React.PropTypes.number,
+    mouseY: React.PropTypes.number,
+    // Provided by redux
+    isInDesignMode: React.PropTypes.bool.isRequired
+  },
+
+  getInitialState: () => ({
+    hoveredControlId: null
+  }),
+
+  componentDidMount() {
+    document.addEventListener('mousemove', this.onMouseMove);
+  },
+
+  componentWillUnmount() {
+    document.removeEventListener('mousemove', this.onMouseMove);
+  },
+
+  onMouseMove(event) {
+    this.setState({
+      hoveredControlId: this.getHoveredControlId(event.target)
+    });
+  },
+
+  /**
+   * Gets the element id of the Applab UI control user is hovering over, if any.
+   * If the user is in design mode, we strip the element id prefix.
+   * @param {EventTarget} eventTarget The mouseover event target
+   * @returns {string} id of the Applab UI control the mouse is over. Returns null if none exist.
+   * @private
+   */
+  getHoveredControlId(eventTarget) {
+    // Check that the element is a child of a screen
+    if (eventTarget && $(eventTarget).parents('div.screen').length > 0) {
+      var controlElement = eventTarget;
```

Comment:
Seems a little confusing to effectively rename a variable. I'd personally just make `controlElement` the name of the method parameter. Your call though.